crates/ide/src/inlay_hints.rs

```diff
@@ -299,6 +305,26 @@ fn is_argument_similar_to_param_name(
     }
 }
 
+fn is_param_name_similar_to_fn_name(
+    param_name: &str,
+    param_num: usize,
+    fn_name: Option<&String>,
+) -> bool {
+    // if its the first parameter, don't show it if:
+    // - is the same as the function name, or
+    // - the function ends with '_' + param_name
+
+    match (param_num, fn_name) {
+        (0, Some(function)) => {
+            function == param_name
+                || (function.len() > param_name.len()
+                    && function.chars().nth(function.len() - param_name.len() - 1) == Some('_')
```

Comment:
```suggestion
                    && function[..function.len() - param_name.len()].ens_with('_')
```

Reply:
Amazing! And please don't worry, I'm the one wasting your time on a feature people are able to live without.

Reply:
You could also replace the entire right hand side of the or-condition with
```rs
function.rfind('_').map(|p| &function[(p + 1)..] == param_name).unwrap_or(false)
```
I think

Reply:
@Veykril That wouldn't work with a parameter with an underscore in the name. See the updated test case.

Reply:
Oh right, I completely forgot that 😅